### PR TITLE
Building for Windows using MSYS2 building platform

### PR DESCRIPTION
**Pull request by mvov**
- Add build scripts for MSYS2.
- Fix NSIS scripts source files.
- Add msys2/README.md with build instruction.
- Update README.md - add reference to build instruction for MSYS2.

### CODE REVIEW

File: msys2/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Installing MSYS2
+Download the installer from [official site](https://www.msys2.org/) or [GitHub Releases page](https://github.com/msys2/msys2-installer/releases).
+
+Run the installer. When done, run "MSYS2 MSYS" from Start menu.  
+
+Update the package database and base packages:
+
+```bash
+pacman -Syu
```
Comment (curtishall) [inline msys2/README.md:11]: @mvov Does msys2 install pacman by default?

File: win/installer32.nsi.in [hunk 2]
```diff
@@ -37,14 +37,14 @@ Section
 !include dll_list.nsh
 
         SetOutPath "$INSTDIR\platforms"
-        File platforms/qdirect2d.dll
-        File platforms/qminimal.dll
-        File platforms/qoffscreen.dll
-        File platforms/qwindows.dll
+        File platforms\qdirect2d.dll
```
Comment (antonsviridenko) [inline win/installer32.nsi.in:40]: Looks like these reverse slash changes are going to break MXE build on Linux hosts. Maybe it's better to make a separate copy of NSIS script with Windows-style path delimiters?

**Review by antonsviridenko — CHANGES_REQUESTED** [review]

File: msys2/README.md (hunk 1 above)
Comment (mvov) [reply msys2/README.md:11]: Yes, pacman is available from the box.

File: win/installer32.nsi.in (hunk 2 above)
Comment (mvov) [reply win/installer32.nsi.in:40]: @curtishall Do we need support both MXE and MSYS2 build?

Comment (curtishall) [reply win/installer32.nsi.in:40]: @antonsviridenko at the moment I'm more interested in spending development time on adding features (HLS) then I am about different build environments.  msys2 seems to work well enough for now.

Comment (antonsviridenko) [reply win/installer32.nsi.in:40]: yeah, but it is not reasonable to break things that work, and it is easy to make a copy of one file, so both build methods can work without issues

**Pull request merged**